src/jit/assertionprop.cpp

```diff
@@ -2995,6 +2995,23 @@ AssertionIndex Compiler::optGlobalAssertionIsEqualOrNotEqual(ASSERT_VALARG_TP as
         {
             return assertionIndex;
         }
+
+        // Look for matching exact type assertions based on vtable accesses
+        if ((curAssertion->assertionKind == OAK_EQUAL) && (curAssertion->op1.kind == O1K_EXACT_TYPE) &&
+            op1->OperIs(GT_IND))
```

Comment:
You should update the method header and describe this new tree that we are matching.

Reply:
Will do.